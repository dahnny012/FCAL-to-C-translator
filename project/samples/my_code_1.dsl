/* This computes matrix multiplication */

main () {
	
	Int maxIndex;
	maxIndex = 5;

	Matrix base = readMatrix ( "../samples/my_data_AA.dat" ) ;
	Matrix data = readMatrix ( "../samples/my_data_AB.dat" ) ;
    Matrix buffer[ maxIndex, maxIndex ] i, j = 
	let
		Int k;
		k = 0;
	in
		k
	end;
	
	
	Int i;
	Int j;
	Int k;
	
	for(i=0 : maxIndex-1){
		for(j=0 : maxIndex-1){
			for(k=0 : maxIndex-1){
				buffer[i,j] = buffer[i,j] + base[i,k] * data[k,j];
			}
		}
	}
	
	print(buffer);
}


