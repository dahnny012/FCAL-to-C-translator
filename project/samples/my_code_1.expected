5 5
15  20  25  30  35  
15  20  25  30  35  
15  20  25  30  35  
15  20  25  30  35  
15  20  25  30  35  
